src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,438 @@
+                System.out.println("  [T][" + status + "] " + tasks.get(i).getDescription());
+            }
+            tasks.remove(i);
+            listCount--;
+            return listCount;
+        }
+        catch (IndexOutOfBoundsException e) {
+            System.out.println("Invalid task number. No items are deleted.");
+        }
+        return listCount;
+    }
+
+    public static void flagAsDone(String input, ArrayList<Task> tasks) {
+        final int IS_DONE_OFFSET = 5;
+        final int ARRAY_OFFSET = 1;
+        final int TODO = 3;
+        final int EVENT = 2;
+        final int DEADLINE = 1;
+        final int TRADITIONAL_TASK = 0;
+        int i = 0;
+        int lastNrPosition = input.length();
+        String sub = input.substring(IS_DONE_OFFSET, lastNrPosition);
+        try {
+            i = Integer.parseInt(sub) - ARRAY_OFFSET;
+            boolean alreadyDone = tasks.get(i).getStatus();
+            if (alreadyDone) {
+                System.out.println("This task is already done!");
+                return;
+            }
+        } catch (IndexOutOfBoundsException e) {
+            //Does not show message since the function will continue to run to the bottom
+        } catch (NumberFormatException e) {
+            System.out.println("Please Enter a number!");
+            return;
+        }
+        try {
+            int taskType = tasks.get(i).getTaskType();
+            System.out.println("Nice! I've marked this task as done: ");
+            if (taskType == TRADITIONAL_TASK) {
+                System.out.println("  [" + "\u2713" + "] " + tasks.get(i).getDescription());
+            } else if (taskType == DEADLINE) {
+                String temp = tasks.get(i).getTime();
+                System.out.println("  [D][" + "\u2713" + "] " + tasks.get(i).getDescription() + "(by:" + temp + ")");
+            } else if (taskType == EVENT) {
+                String temp = tasks.get(i).getTime();
+                System.out.println("  [E][" + "\u2713" + "] " + tasks.get(i).getDescription() + "(on:" + temp + ")");
+            } else if (taskType == TODO) {
+                System.out.println("  [T][" + "\u2713" + "] " + tasks.get(i).getDescription());
+            }
+            tasks.get(i).markAsDone();
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println("Invalid task number or task does not exist. Please try again.");
+        }
+    }
+
+    public static void displayList(int listCount, ArrayList<Task> tasks) {
+        final int ARRAY_OFFSET = 1;
+        int i;
+        System.out.println("Here are the tasks in your list:");
+        for (i = 0; i < listCount; i++) {
+            if (tasks.get(i).getDescription() != null) {
+                System.out.println(i + ARRAY_OFFSET + ". " + tasks.get(i).toString());
+            }
+        }
+    }
+
+    public static int createTraditionalTask(String input, int listCount, ArrayList<Task> tasks, boolean initialize)
```

Comment:
consider handling this in the Task class, Likewise for the other similar functions below